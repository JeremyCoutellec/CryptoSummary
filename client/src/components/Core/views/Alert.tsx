import React, { Fragment } from 'react';
import map from 'lodash/map';

// Redux
import { useSelector } from 'react-redux';
import { getCoreAlertSelector } from '../selectors/auth';

// Material UI
import AlertMaterial from '@mui/material/Alert';
import AlertTitle from '@mui/material/AlertTitle';

const Alert = () => {
    const alerts = useSelector(getCoreAlertSelector);
    return (
        <Fragment>
            {map(alerts, (alert, index) => (
                <AlertMaterial key={index} severity={alert.alertType} className="alertBox">
                    {alert?.title && <AlertTitle>{alert?.title}</AlertTitle>}
                    {alert.msg}
                </AlertMaterial>
            ))}
        </Fragment>
    );
};

export default Alert;
