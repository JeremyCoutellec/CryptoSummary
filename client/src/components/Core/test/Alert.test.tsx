import React from 'react';
import { Provider } from 'react-redux';
import createStore from '../../../store';
import renderer from 'react-test-renderer';
import Alert from '../views/Alert';

let store;
describe('Your test', () => {
    beforeEach(() => {
        store = createStore;
    });
    test('Alerts rendering', () => {
        const component = renderer.create(
            <Provider store={store}>
                <Alert />,
            </Provider>
        );
        const alert = component.toJSON();
        expect(alert).toMatchSnapshot();
    });
});
