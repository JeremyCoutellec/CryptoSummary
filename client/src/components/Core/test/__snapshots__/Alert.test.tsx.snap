// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Your test Alerts rendering 1`] = `","`;
